File: mobile/src/main/java/com/example/steven/stlbusarrivals/VolleySingleton.java
package com.example.steven.stlbusarrivals;

import android.content.Context;

import com.android.volley.RequestQueue;
import com.android.volley.toolbox.Volley;

/**
 * Created by Steven on 2016-01-22.
 */
public class VolleySingleton {

    private static VolleySingleton instance;
    private RequestQueue requestQueue;

    private VolleySingleton(Context context) {
        requestQueue = Volley.newRequestQueue(context);
    }

    public static VolleySingleton getInstance(Context context) {
        if (instance == null) {
            instance = new VolleySingleton(context);
        }
        return instance;
    }

    public RequestQueue getRequestQueue() {
        return requestQueue;
    }
}

File: mobile/src/main/java/com/tilly/steven/stlbusarrivals/VolleySingleton.kt
package com.tilly.steven.stlbusarrivals

import android.content.Context

import com.android.volley.RequestQueue
import com.android.volley.toolbox.Volley

/**
 * Created by Steven on 2016-01-22.
 */
class VolleySingleton private constructor(context: Context) {
    val requestQueue: RequestQueue = Volley.newRequestQueue(context)

    companion object {

        private var instance: VolleySingleton? = null

        fun getInstance(context: Context): VolleySingleton {
            if (instance == null) {
                instance = VolleySingleton(context)
            }
            return instance!!
        }
    }
}
